### ISSUE TITLE
TypeError: Unknown StorageFactory when testing in addon

### ISSUE THREAD

**Issue by BryanCrotaz** [funkensturm/ember-local-storage#259]
Using storageFor in addon fails at test time with error
TypeError: Unknown StorageFactory: storage:<name>
    at createStorage

**Comment by fsmanuel** [comment]
@BryanCrotaz thanks for reporting. Is the addon public?

**Comment by BryanCrotaz** [comment]
No, but you should be able to reproduce by creating a service in an addon and having the service use storageFor and get it in init

**Comment by fsmanuel** [comment]
Did you define the storage object in `app/storages` or `addon/storages`?

**Comment by BryanCrotaz** [comment]
used your generator which puts it in app

**Comment by fsmanuel** [comment]
Will dig into it.

**Comment by fsmanuel** [comment]
How do you test it? I mean do you use it in the dummy app or use the addon in a normal app? Would be great if you could provide a repo to reproduce the error.

**Comment by fsmanuel** [comment]
@BryanCrotaz any news?

**Comment by tinyweasel** [comment]
Any news on this? I'd like to know how to make the tests work.

**Comment by fsmanuel** [comment]
@tinyweasel can you provide a repo to reproduce?

**Comment by fsmanuel** [comment]
@BryanCrotaz @tinyweasel If one of you can provide a repo to reproduce I'm happy to fix it.

**Comment by carlos-figueroa-globant** [comment]
Hi, any updates with this issue?

**Comment by fsmanuel** [comment]
@carlos-figueroa-globant I still can not reproduce it. If you have a repo I can test with I'll take a look.

**Comment by vitaliy-petrenko** [comment]
just copy storages into addon folder and make in app/storages/storage-name:

export { default } from 'repo-name/storages/storage-name';

like other generators do this

**Comment by ammorrise** [comment]
For anyone who may be struggling with this: We discovered that adding the storage key to the `needs` array in your test seems to solve the problem.

In the route/controller:
```
myModel: storageFor('my-model')
```

In the test:
```
needs: [
    'storage:my-model'
]
```

**Comment by fsmanuel** [comment]
@ammorrise Thank you for that hint!
